site/src/Consumer/EmailNotificationConsumer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Consumer;
+
+use FOS\UserBundle\Model\UserManagerInterface;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+use App\Util\NotificationHydrator;
+
+/**
+ * Class EmailNotificationConsumer
+ *
+ * This class is responsible for sending out Notifications
+ * to the user's email
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+class EmailNotificationConsumer extends AbstractNotificationConsumer implements ConsumerInterface
+{
+    /**
+     * EmailNotificationConsumer constructor.
+     * @param UserManagerInterface $userManager
+     */
+    public function __construct(UserManagerInterface $userManager)
+    {
+        parent::__construct($userManager);
+    }
+
+    /**
+     * Find the user that created the notification
+     * and persist it in the database
+     *
+     * @param AMQPMessage $msg
+     *
+     * @throws \Exception
+     */
+    public function execute(AMQPMessage $msg)
+    {
+        echo "\nEmail Notification Consumer\n";
+    }
```

Comment:
Can you put the implementation of `execute()` into `AbstractNotificationConsumer`, and add another abstract method `handle()` which is then implemented in the child classes?